common/src/models/project.rs

```diff
@@ -17,6 +18,14 @@ pub const fn idle_minutes() -> u64 {
     IDLE_MINUTES
 }
 
+#[derive(Debug, Clone, Copy, Deserialize, IntoParams)]
+pub struct PaginationDetails {
```

Comment:
Ah, yep. Close to what I meant but I was thinking about having a single common struct that can be shared between deployments and projects list endpoints. The thing is that I am not really sure where it can be put, here is one thought thread:
1) Create `common/src/api_query`.
2) Create the `PaginationDetails` struct under `common/src/backends/api_query.rs` and consume it from there.
3) Curious if you have thoughts on creating `common/src/api_query/mod.rs` instead of `common/src/backends/api_query.rs`. I would go with 2) now rather than 3) because 3) makes sense only if there are other `api_query` functionality that can be split between multiple sub-module parts that can target specificities with `api_query` (e.g. for deployments, projects, and others).
4) Maybe another idea?

Let me know what you think.

Reply:
The issue with a shared struct is that if we end up adding filtering capabilities to `project list` (the third bullet point [here](https://github.com/shuttle-hq/shuttle/issues/845#issue-1692726868), the filters would also be shared between both commands, which I think is not ideal. We talked about having a common struct that contains _only_ the pagination data, and then `#[serde(flatten)]` any additional parameters, but we decided that it's not worth the additional layer of abstraction and that it's better to just keep 2 separate structs around. What do you think? For reference, we are working on the filters on [this branch](https://github.com/Kazy/shuttle/tree/feat/project-filtering).

Reply:
Let me know if this is accurate: your suggestion is for having two separate structs called `PaginationDetails`, one for project and one for deployment, and the project filters would be added as additional struct fields on the project's `PaginationDetails` struct. This should work as well. 

Your suggestion for `#[serde(flatten)]` opened me up for an example regarding pagination, present in the official docs. My thinking was along the below lines: 

```rust
#[derive(Serialize, Deserialize)]
struct PaginationDetails {
    limit: u64,
    offset: u64
}

#[derive(Serialize, Deserialize)]
struct ProjectFilter {
    state: State,

    #[serde(flatten)]
    pagination: PaginationDetails,
}
```

So instead of having the shared pagination struct with the pagination fields + additional `#[serde(flatten)]` parameters, we can have the queries for each resource defined in their own struct adding the `PaginationDetails` with `#[serde(flatten)]`.

Using such a layout for the struct can be easily mapped to my second point [here](https://github.com/shuttle-hq/shuttle/pull/862#discussion_r1185996913). Either way, I do not have a strong preference right now about the `PaginationDetails` being a single struct versus two separate structs.

I am curious how everything will look like once you add project filtering. 😄

Reply:
Yes, this was pretty much what I had in mind as well. However, I think it's an unnecessary abstraction and unless we intend to add this functionality to other commands as well (such as `cargo shuttle resource`, I don't think it's worth doing.

Reply:
Sure, there is not a mandatory requirement on this one given the struct is duplicated only for two places. We can refactor this in the future if needed. Resolving this as well.